File: 1742-maximum-number-of-balls-in-a-box/1742-maximum-number-of-balls-in-a-box.py
class Solution:
    def countBalls(self, l: int, h: int) -> int:
        mp = [0] * 46
        for i in range(l,h+1):
            som,n = 0,i
            while n > 0:
                som += n % 10
                n =n // 10
            mp[som]+=1
        return max(mp)
            

File: 1742-maximum-number-of-balls-in-a-box/1742-maximum-number-of-balls-in-a-box.cpp
class Solution {
public:
    int countBalls(int l, int h) {
        int map[46] = {};
        for(int i = l; i <= h; i++)
        {
            int sum(0),n(i);
            while(n > 0)
            {
                sum += n % 10;
                n /= 10;
            }
            map[sum]++;
        }
        return *max_element(begin(map),end(map));
    }
};